test/extensions/tracers/opentelemetry/span_context_extractor_test.cc

```diff
@@ -144,6 +144,27 @@ TEST(SpanContextExtractorTest, ThrowsExceptionWithAllZeroParentId) {
   EXPECT_THAT(span_context, HasStatusMessage("Invalid parent id"));
 }
 
+TEST(SpanContextExtractorTest, ExtractSpanContextWithEmptyTracestate) {
+  Http::TestRequestHeaderMapImpl request_headers{
+      {"traceparent", fmt::format("{}-{}-{}-{}", version, trace_id, parent_id, trace_flags)}};
+  SpanContextExtractor span_context_extractor(request_headers);
+  absl::StatusOr<SpanContext> span_context = span_context_extractor.extractSpanContext();
+
+  EXPECT_OK(span_context);
+  EXPECT_TRUE(span_context->tracestate().empty());
+}
+
+TEST(SpanContextExtractorTest, ExtractSpanContextWithTracestate) {
+  Http::TestRequestHeaderMapImpl request_headers{
+      {"traceparent", fmt::format("{}-{}-{}-{}", version, trace_id, parent_id, trace_flags)},
+      {"tracestate", "sample-tracestate"}};
```

Comment:
Is there one for missing traceparent?

Reply:
Ah there isn't, good call! Technically it was already covered via the missing traceparent test above, but I think it's worthwhile to explicitly add one for it for legibility. I also fleshed out the comment in the extractor, mentioning that we are already checking for the existence of the traceparent header.